File: backend/springboot/src/main/java/com/darkcandles/springboot/product/Product.java
package com.darkcandles.springboot.product;

public abstract class Product {
  public Integer ProductId;
  public Character ProductType;
  public String Description;
  public Integer Stock;
  public Double Price;

  public Product(Integer productId, Character productType, String description, Integer stock, Double price) {
    ProductId = productId;
    ProductType = productType;
    Description = description;
    Stock = stock;
    Price = price;
  }

  public Integer getProductId() {
    return ProductId;
  }

  public void setProductId(Integer productId) {
    ProductId = productId;
  }

  public Character getProductType() {
    return ProductType;
  }

  public void setProductType(Character productType) {
    ProductType = productType;
  }

  public String getDescription() {
    return Description;
  }

  public void setDescription(String description) {
    Description = description;
  }

  public Integer getStock() {
    return Stock;
  }

  public void setStock(Integer stock) {
    Stock = stock;
  }

  public Double getPrice() {
    return Price;
  }

  public void setPrice(Double price) {
    Price = price;
  }
}

File: src/app/product-page/Product.ts
export abstract class Product {
  private _id: string;
  private _productType: string;
  private _description: string;
  private _stock: number;
  private _price: number;
  private _created: Date;

  protected constructor(id: string, productType: string, description: string, stock: number, price: number, created: Date) {
    this._id = id;
    this._productType = productType;
    this._description = description;
    this._stock = stock;
    this._price = price;
    this._created = created;
  }

  getStock(): string {
    return String(this._stock);
  }

  getPrice(): string {
    return String(this._price);
  }

  getId(): string {
    return this._id;
  }

  getProductType(): string {
    return this._productType;
  }

  getDescription(): string {
    return this._description;
  }

  getCreated(): Date {
    return this._created;
  }
}
